### PR TITLE
[Bug] [Api] Not refresh workflow list when batch delete workflow failed

### LINKED ISSUE #10674: [Bug] [Api]  Not refresh workflow list when batch delete workflow failed
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar issues.

### What happened

batch delete workflow failed, workflow list not refresh
image

### What you expected to happen

Refreshes the workflow list regardless of whether the batch delete is successful or not

### How to reproduce

create two workflows,  one is online and other is offline, then batch delete all of them.

### Anything else

_No response_

### Version

dev

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by zhuxt2015**
## Purpose of the pull request
fix #10674

## Verify this pull request

This pull request is already covered by existing tests, such as *(please describe tests)*.

### CODE REVIEW

File: dolphinscheduler-ui/src/views/projects/workflow/definition/use-table.ts [hunk 1]
```diff
@@ -264,6 +264,12 @@ export function useTable() {
         pageNo: variables.page,
         searchVal: variables.searchVal
       })
+    }).catch(() => {
+      getTableData({
+        pageSize: variables.pageSize,
+        pageNo: variables.page,
+        searchVal: variables.searchVal
+      })
```
Comment (Amy0104) [inline dolphinscheduler-ui/src/views/projects/workflow/definition/use-table.ts]: If the page is two with only one record, there will have a bug after deleting a workflow.

File: dolphinscheduler-ui/src/views/projects/workflow/definition/use-table.ts [hunk 2]
```diff
@@ -257,6 +257,20 @@ export function useTable() {
...
+    }).catch(() => {
+      if (
+          Math.ceil((variables.columns.length - variables.checkedRowKeys.length) / variables.pageSize) < variables.page
+          && variables.page > 1
+      ) {
+        variables.page -= 1;
+      }
+
```
Comment (Amy0104) [inline dolphinscheduler-ui/src/views/projects/workflow/definition/use-table.ts:273]: There are still some problems. 

-  Such as the current page is 1.
-  Why using columns length divide checkedRowKeys length ?
-  It's better to set a common function to handle refresh table.

**Review by Amy0104 — CHANGES_REQUESTED** [review]

File: dolphinscheduler-ui/src/views/projects/workflow/definition/use-table.ts [hunk 3]
```diff
@@ -257,6 +257,20 @@ export function useTable() {
...
       variables.checkedRowKeys = []
```
Comment (Amy0104) [inline dolphinscheduler-ui/src/views/projects/workflow/definition/use-table.ts:347]: There are still some problems:
- The current page is 1
- Why using columns length divide checkedRowKeys length
- It is better to set a common function to refresh table

**Review by Amy0104 — CHANGES_REQUESTED** [review]

File: dolphinscheduler-ui/src/views/projects/workflow/definition/use-table.ts (hunk 3 above)
Comment (zhuxt2015) [reply dolphinscheduler-ui/src/views/projects/workflow/definition/use-table.ts:347]: If delete some workflow in page 1 and all workflow of page 2,  and current page is 2,  after delete operation, current page will switch to page 1.  "columns length divide checkedRowKeys length" resolve this.

Comment (zhongjiajie) [reply dolphinscheduler-ui/src/views/projects/workflow/definition/use-table.ts:347]: Do we still have discussion here? @zhuxt2015 @Amy0104

Comment (Amy0104) [reply dolphinscheduler-ui/src/views/projects/workflow/definition/use-table.ts:347]: Columns length means how many columns that the data table has, and it does not seem to be related to how many workflows that the data table has. If it is partially deleted successfully, it is hardly to know the current page number from the front end part.

**Comment by EricGao888** [comment]
@zhuxt2015 Hi, may I ask whether you are still interested in following up with this PR? Thanks : )

**Review by songjianet — APPROVED** [review]

**Comment by rickchengx** [comment]
It seems that there is still an unresolved comment, cc @songjianet @zhuxt2015 @Amy0104

**Comment by SbloodyS** [comment]
Close due to inactivity.

**Pull request closed without merging**